consul/consul.go

```diff
@@ -232,6 +242,17 @@ func (l *Lease) Renew(ctx context.Context) error {
 
 // Close destroys the underlying session.
 func (l *Lease) Close() error {
+	// Attempt to remove key before destroying session.
+	kvKey := l.leaser.kvKey()
+	if ok, _, err := l.leaser.client.KV().Release(&api.KVPair{
+		Key:     kvKey,
+		Session: l.sessionID,
+	}, nil); err != nil {
+		log.Printf("consul key release error: key=%s session=%s", kvKey, l.sessionID)
+	} else if !ok {
+		log.Printf("cannot release consul key: key=%s session=%s", kvKey, l.sessionID)
+	}
+
```

Comment:
`Client.Session().Destroy()` does not remove the key/value pair associated with the session. This `Release()` call will clear the key while also removing the lock on it.